azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -333,25 +361,40 @@ private boolean isAlreadyRunning(final int execId) throws ExecutorManagerExcepti
     return false;
   }
 
+  private boolean isExecutorSpecified(final ExecutableFlow flow) {
+    return flow.getExecutionOptions().getFlowParameters()
+        .containsKey(ExecutionOptions.USE_EXECUTOR);
+  }
+
   private FlowRunner createFlowRunner(final int execId) throws ExecutorManagerException {
     final ExecutableFlow flow;
     flow = this.executorLoader.fetchExecutableFlow(execId);
     if (flow == null) {
       throw new ExecutorManagerException("Error loading flow with exec "
           + execId);
     }
-
+    
     // Record the time between submission, and when the flow preparation/execution starts.
     // Note that since submit time is recorded on the web server, while flow preparation is on
     // the executor, there could be some inaccuracies due to clock skew.
     commonMetrics.addQueueWait(System.currentTimeMillis() -
         flow.getExecutableFlow().getSubmitTime());
 
     final Timer.Context flowPrepTimerContext = commonMetrics.getFlowSetupTimerContext();
+
+    // Sets up the project files and execution directory.
+    this.preparingFlowCount.incrementAndGet();
     try {
-      // Sets up the project files and execution directory.
-      this.flowPreparer.setup(flow);
+      if (this.active || isExecutorSpecified(flow)) {
```

Comment:
Please add some comments for isExecutorSpecified(flow).

Reply:
added, thanks